run_jobs.py

```diff
@@ -0,0 +1,135 @@
+
+    def create_metadata(self) -> None:
+        started_at_dt = pd.to_datetime(self.started_at, unit='s')
+        finshed_at_dt = pd.to_datetime(self.finished_at, unit='s')
+
+        job_run_df = pd.DataFrame({
+            'job_name': [self.name],
+            'started_at': [started_at_dt],
+            'finished_at': [finshed_at_dt]
+        })
+        job_run_df.to_sql('job_run', db_creator_obj.engine, if_exists='append', index=False)
+        logger.info(f'Inserted job_run record with finished_at value of {started_at_dt}')
+        job_run_id = pd.read_sql('job_run', db_creator_obj.engine).iloc[-1]['id']
+
+        if len(self.data_sources) == 0:
+            logger.warning('No valid data sources were identified')
+        else:
+            data_source_status_df = pd.DataFrame(self.data_sources)
+            data_source_status_df = data_source_status_df.assign(**{'job_run_id': job_run_id})
+            data_source_status_df.to_sql('data_source_status', db_creator_obj.engine, if_exists='append', index=False)
+            logger.info(f'Inserted {len(data_source_status_df)} data_source_status records')
+
+    def run(self) -> Dict[str, str]:
+        leaf_module = import_module(self.import_path)
+        start_method = getattr(leaf_module, self.method_name)
+
+        # Until we have a decorator for this
+        self.started_at = time.time()
+        data_sources = start_method()
+        self.finished_at = time.time()
+
+        delta = self.finished_at - self.started_at
+        str_time = time.strftime('%H:%M:%S', time.gmtime(delta))
+        logger.info(f'Duration of job run: {str_time}')
+
+        valid_data_sources = []
+        for data_source in data_sources:
+            data_source_name = data_source['data_source_name']
+            try:
+                valid_data_source_name = ValidDataSourceName[data_source_name]
+                valid_data_sources.append(data_source)
+            except KeyError:
+                logger.error(f'{data_source_name} is not a valid data source name')
+                logger.error(f'No data_source_status record will be inserted.')
+
+        self.data_sources = valid_data_sources
+        self.create_metadata()
+
+
+class JobManager:
+
+    def __init__(self, job_names: Sequence[str]) -> None:
+        self.jobs = []
+        for job_name in job_names:
+            try:
+                valid_job_name = ValidJobName[job_name.upper()]
+                self.jobs.append(Job(valid_job_name))
+            except KeyError:
+                logger.error(f'{job_name} is not a valid job name')
+
+    def run_jobs(self):
+        for job in self.jobs:
+            logger.info(f'- - Running job {job.name} - -')
+            job.run()
+
+
+if __name__ == '__main__':
+    how_started = os.environ.get('HOW_STARTED', None)
+
+    if how_started == 'DOCKER_COMPOSE':
+        logger.info('Waiting for the MySQL turtle, hehe')
+        # Wait for MySQL container to finish setting up
+        time.sleep(30.0)
```

Comment:
@jonespm, I had to add this because on the first startup (when there is no `.data` directory), it takes a while to finish setting up and the subsequent `db_creator_obj.migrate()` call was failing. Is there a way to specify in `docker-compose` that the job container needs to wait for the MySQL one? This was just a hack to get things working. Also, it's annoying because the first time you need about 30 seconds, but if the `.data` directory exists, it takes about 3 seconds.

Reply:
I'm doing this in the MyLA start script with the nc command which should be installed by default. 
```
echo "Waiting for DB"
while ! nc -z ${MYSQL_HOST} ${MYSQL_PORT}; do   
  sleep 1 # wait 1 second before check again
done
```

I believe the we could have a shell script before this python or could just do this similar check in python with nc. (I didn't test this code below)

```
while not os.system(f"nc -z {os.getenv("MYSQL_HOST")} {os.getenv("MYSQL_PORT")}):
    time.sleep (1)
```

Reply:
Okay, I'll work on this now.

Reply:
Kinda remembered something like this, forgot to check MyLA

Reply:
I created a new issue, #67.